### ISSUE TITLE
NSS: Introduce underscore-version variable

### ISSUE THREAD

**Issue by msirringhaus** [openSUSE/firefox-maintenance#1]
Antonio noticed that there is the possibility of an inconsitency in the spec-file of NSS (it actually was there):
The source-URL is something like this:
`https://ftp.mozilla.org/pub/mozilla.org/security/nss/releases/NSS_3_44_RTM/src/nss-%{version}.tar.gz`

If the version is changed, there is a good chance that the hardcoded underscore version in the URL is missed to be updated.

If we would define something like
`%define underscore_version %(echo "%{version}" | tr . _)`
this could be automated.

**Comment by wrosenauer** [comment]
I would take this and will incorporate for Factory first if you agree.

**Comment by msirringhaus** [comment]
ok

**Comment by wrosenauer** [comment]
%define         underscore_version %(echo "%{version}" | tr . _)
Source:         https://ftp.mozilla.org/pub/mozilla.org/security/nss/releases/NSS_%{underscore_version}_RTM/src/nss-%{version}.tar.gz

but

wolfi@Hygiea:/local/build/mozilla/mozilla-nss> osc service localrun download_files
ERROR: Failed to download "https://ftp.mozilla.org/pub/mozilla.org/security/nss/releases/NSS_MACRO_RTM/src/nss-3.44.tar.gz"

Did  I make a mistake or is the osc service unable to expand the macro?

**Comment by msirringhaus** [comment]
So, I played a bit with that. 
If I do 
```
%prep
echo "https://ftp.mozilla.org/pub/mozilla.org/security/nss/releases/NSS_%{underscore_version}_RTM/src/nss-%{version}.tar.gz"
exit 1
```
it echos the correct string. So I guess, osc can't do that, which is a shame.

Second best solution might be to do 
```
%define underscore_version 3_44
Version:                   3.44
```
instead.. 
Not as nice, but it would minimize the possibility of overlooking it in the URL.

**Comment by wrosenauer** [comment]
Done in mozilla:Factory/mozilla-nss

(Process-wise I'm not sure if I'm going to close this.)
This package will be submitted to Factory/Tumbleweed soon. When it will end up in Leap depends on when FF68esr hits those.

**Comment by msirringhaus** [comment]
You could do that in this repo here as well and push it? Then we can see if you have enough permissions to actually work on here.
If you mention something like "Fix #1" in the commit message, this issue will close automatically.
